### PR TITLE
[#9764] Add instructors as respondents in remind modal

### LINKED ISSUE #9764: Feedback Session Page: Remind modal not show instructor involved
- **Environment**: `master` branch at commit a1ca379b54ad2ab2d2a509d54129809a61605dc6

**Steps to reproduce**
1. log in as instructor
2. create a session contains questions for both student and instructor to answer
3. in the session table, find the session created and click `Remind` button under Action(s)

**Actual behaviour**

Only students involved in this session are shown in the modal, but no instructors showed.

**Expected behaviour**
Should also include the instructors involved for this session to remind.

### PR DESCRIPTION
**Pull request by madanalogy**
Fixes #9764, Closes #9788 

**Outline of Solution**

Refactored _students_ in relevant data structures and components to _respondents_ to include _instructors_

### CODE REVIEW

File: src/web/app/components/sessions-table/resend-results-link-to-respondent-modal/resend-results-link-to-respondent-modal.component.spec.ts [hunk 1]
```diff
@@ -3,22 +3,22 @@ import { Component } from '@angular/core';
...
 import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
-import { StudentListInfoTableComponent } from '../student-list-info-table/student-list-info-table.component';
-import { ResendResultsLinkToStudentModalComponent } from './resend-results-link-to-student-modal.component';
+import { RespondentListInfoTableComponent } from '../respondent-list-info-table/respondent-list-info-table.component';
+import { ResendResultsLinkToRespondentModalComponent } from './resend-results-link-to-respondent-modal.component';
 
 @Component({ selector: 'tm-ajax-preload', template: '' })
 class AjaxPreloadComponent {}
 
 describe('ResendResultsLinkToStudentModalComponent', () => {
```
Comment (t-cheepeng) [inline src/web/app/components/sessions-table/resend-results-link-to-respondent-modal/resend-results-link-to-respondent-modal.component.spec.ts:12]: The name here should change to `ResendResultsLinkToRespondentModalComponent`

File: src/web/app/components/sessions-table/send-reminders-to-respondents-modal/send-reminders-to-respondents-modal.component.spec.ts [hunk 2]
```diff
@@ -4,22 +4,22 @@ import { HttpClientTestingModule } from '@angular/common/http/testing';
...
 import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
-import { StudentListInfoTableComponent } from '../student-list-info-table/student-list-info-table.component';
-import { SendRemindersToStudentModalComponent } from './send-reminders-to-student-modal.component';
+import { RespondentListInfoTableComponent } from '../respondent-list-info-table/respondent-list-info-table.component';
+import { SendRemindersToRespondentsModalComponent } from './send-reminders-to-respondents-modal.component';
 
 @Component({ selector: 'tm-ajax-preload', template: '' })
 class AjaxPreloadComponent {}
 
 describe('SendRemindersToStudentModalComponent', () => {
```
Comment (t-cheepeng) [inline src/web/app/components/sessions-table/send-reminders-to-respondents-modal/send-reminders-to-respondents-modal.component.spec.ts:13]: same here

File: src/web/app/components/sessions-table/send-reminders-to-respondents-modal/send-reminders-to-respondents-modal.component.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  }
+
+  ngOnInit(): void {
+  }
+
+  /**
+   * Changes selection state for all students.
+   */
+  changeSelectionStatusForAllStudentsHandler(shouldSelect: boolean): void {
+    for (const model of this.studentListInfoTableRowModels) {
```
Comment (ccyccyccy) [inline src/web/app/components/sessions-table/send-reminders-to-respondents-modal/send-reminders-to-respondents-modal.component.ts:34]: Use `forEach` here?

File: src/web/app/components/sessions-table/send-reminders-to-respondents-modal/send-reminders-to-respondents-modal.component.ts [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    for (const model of this.instructorListInfoTableRowModels) {
+      model.isSelected = shouldSelect;
+    }
+  }
+
+  /**
+   * Changes selection state for all yet to submit instructors.
+   */
+  changeSelectionStatusForAllYetSubmittedInstructorsHandler(shouldSelect: boolean): void {
+    for (const model of this.instructorListInfoTableRowModels) {
```
Comment (ccyccyccy) [inline src/web/app/components/sessions-table/send-reminders-to-respondents-modal/send-reminders-to-respondents-modal.component.ts:63]: Until here

File: src/main/java/teammates/ui/webapi/action/RemindFeedbackSessionSubmissionAction.java [hunk 5]
```diff
@@ -40,8 +40,8 @@ public ActionResult execute() {
                     + "as the feedback session is not open for submissions.", HttpStatus.SC_BAD_REQUEST);
         }
 
-        FeedbackSessionStudentRemindRequest remindRequest =
```
Comment (xpdavid) [inline src/main/java/teammates/ui/webapi/action/RemindFeedbackSessionSubmissionAction.java:43]: Just checked. We need to add a backend test for instructors (currently only have one for students)

**Pull request merged**